### ISSUE TITLE
isDate/isValidDate returns false when it should be true

### ISSUE THREAD

**Issue by vorticalbox** [char0n/ramda-adjunct#724]
Valid dates are being returned as false.

"ramda": "^0.25.0"
"ramda-adjunct": "^2.10.0"

node 8.12.0 && 9.7.1

```
const ramda = require('ramda')
const RA = require('ramda-adjunct')
console.log(RA.isDate(new Date()))
console.log(RA.isValidDate(new Date()))
console.log(RA.isNotDate(new Date()))
console.log(RA.isNotValidDate(new Date()))
```

```
false
false
true
true
```

https://repl.it/@vorticalbox/GoldenrodLuxuriousValue

**Comment by char0n** [comment]
Thank you for report @vorticalbox. I'll look into this ASAP.

**Comment by char0n** [comment]
Currently not able to reproduce in any browser - chrome, firefox, edge.

Also this [test](https://github.com/char0n/ramda-adjunct/blob/a3109fc1020c6f4a8686d032cb7d539a230af258/test/isDate.js#L8) is running on node 6,7,8 and 9 successfully and without problems.

Trying to investigate further why it's not working in your case.

**Comment by char0n** [comment]
Cannot reproduce the issue on node 8.12.0 || 9.7.1. If I use your code and save in file named `test.js` and  run the file with `node test.js` everything is fine.

IMHO I think that the problem my be in REPL itself. `RA.isDate` is using [R.is](https://ramdajs.com/docs/#is) to check for date instances. JavaScript have prototypal nature and concept of realms. For example if you have two tabs in the browser opened and send a date object from one to another, even though the transferred object is of `Date` instance, if you check it with `transferedDate intanceof Date` it will return false. Both tabs have their own realms and `instanceof` operator only works for one realm, it cannot work cross realms. 

In Ramda-Adjunct we don't try to solve the realm problem, because we consider it a corner case not worth handling.

**Comment by vorticalbox** [comment]
Thanks, the repl.it  I posted has the same issue, also the application I am using it in is in node lts running inside docker.

What's weird is when I just use ramda is function is completely fine.

**Comment by srghma** [comment]
@char0n https://stackoverflow.com/a/643827/3574379

```
A work-around for this is to check the object's class via

Object.prototype.toString.call(date) === '[object Date]'

```

**Comment by char0n** [comment]
Yes exactly, that is what we already do wjth other predicates

**Comment by char0n** [comment]
Fixes are comming in next couple of days in 2.12.0 release

**Comment by char0n** [comment]
@vorticalbox all affected predicates should be fixed now. Checkout the pending PRs if you can. Those PR will be part of v2.13.0

**Comment by char0n** [comment]
Released as v2.13.0. @vorticalbox hopefully your problem will be resolved.

**Comment by vorticalbox** [comment]
Thanks for the update :) I'll look forward to testing it out.
